### DISCUSSION TITLE
Support for linux-drm-syncobj-v1 protocol

### DISCUSSION THREAD

**Discussion by Heus-Sueh** [CuarzoSoftware/Louvre#79] [Ideas]
This protocol fixes almost all flickering, glitches and artifact issues in Wayland on NVIDIA cards.

https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/90

> This protocol will allow explicit synchronization of buffers using DRM synchronization objects. While being a protocol that is unlikely to be widely used directly by applications and toolkits themselves, it is an important building block for improving Vulkan and OpenGL drivers.

**Comment by ehopperdietzel** [comment]
Hi, thanks for the suggestion! I'll try to implement it soon. By the way, are you experiencing that with Louvre? If so, does it happen only with proprietary drivers or with Nouveau as well?

**Comment by Heus-Sueh** [comment]
> Hi, thanks for the suggestion! I'll try to implement it soon. By the way, are you experiencing that with Louvre? If so, does it happen only with proprietary drivers or with Nouveau as well?

I haven't tried it yet, I just suggested it because future Louvre users will benefit from this protocol, since in almost all cases Explict Sync is better

**Comment by Fox2Code** [comment]
The proprietary driver doesn't support implicit sync, `linux-drm-syncobj-v1` is the protocol for explicit sync.

SRM/Louvre somehow worked around issues with the proprietary Nvidia driver in other ways.

**Comment by ehopperdietzel** [comment]
> I haven't tried it yet, I just suggested it because future Louvre users will benefit from this protocol, since in almost all cases Explict Sync is better.

Ok.

> SRM/Louvre somehow worked around issues with the proprietary Nvidia driver in other ways.

I don't think so. With my Nvidia card (which is quite old), I haven't experienced those issues when using proprietary drivers, so I guess it's a thing of newer drivers/cards.

**Comment by Fox2Code** [comment]
Well, as long as stuff work, maybe due to how SRM/Louvre works `linux-drm-syncobj-v1` is not a requirement?

Well I don't think implementing `linux-drm-syncobj-v1` would hurt.

**Comment by ehopperdietzel** [comment]
I read that it should be more efficient than the implicit sync mechanism most drivers implement, so I'll definitely add it 😉

**Comment by Ramblurr** [comment]
I see that the issue was closed and turned into this discussion..does that mean this protocol is supported? I have a newer Nvidia card and could help test.

**Reply by ehopperdietzel** [reply]
Hi, it’s not closed, I haven’t implemented the protocol yet, but I will. Thank you for offering to test it! Are you noticing any glitches when running the examples?
